scalafix-core/src/main/scala/scalafix/rewrite/ExplicitReturnTypes.scala

```diff
@@ -20,18 +22,39 @@ case class ExplicitReturnTypes(mirror: Mirror)
 
   def defnName(defn: Defn): Option[Name] = Option(defn).collect {
     case Defn.Val(_, Seq(Pat.Var.Term(name)), _, _) => name
+    case Defn.Var(_, Seq(Pat.Var.Term(name)), _, _) => name
     case Defn.Def(_, name, _, _, _, _) => name
   }
 
+  def visibility(mods: Traversable[Mod]): MemberVisibility =
+    mods
+      .collect {
+        case _: Mod.Private => MemberVisibility.Private
+        case _: Mod.Protected => MemberVisibility.Protected
+      }
+      .headOption
+      .getOrElse(MemberVisibility.Public)
+
+  def kind(defn: Defn): Option[MemberKind] = Option(defn).collect {
+    case _: Defn.Val => MemberKind.Val
+    case _: Defn.Def => MemberKind.Def
+    case _: Defn.Var => MemberKind.Var
+  }
+
   def parseDenotationInfo(denot: Denotation): Option[Type] = {
     val base =
-      if (denot.isVal) denot.info
+      if (denot.isVal || denot.isVar) denot.info
       else if (denot.isDef) denot.info.replaceFirst(".*\\)", "")
       else {
         throw new UnsupportedOperationException(
           s"Can't parse type for denotation $denot, denot.info=${denot.info}")
       }
-    base.parse[Type].toOption
+    if (denot.isVal || denot.isDef)
+      base.parse[Type].toOption
+    else
+      base.parse[Stat].toOption.flatMap {
```

Comment:
Interesting, it seems the symbol for vars point to the setter. This may change in the future, FYI, but this solution is OK for now! We have tests :)

Can we use a for comprehension instead?

```scala
// NOTE: The symbol for vars points to the getter, this may change in the future
// but for now we parse the structure of the getter's denotation.  
for {
  stat <- base.parse[Stat].toOption
  typ <- stat.collectFirst { case Term.Ascribe(_, typ) => typ }
} yield typ
```